x/tx/textual/dec.go

```diff
@@ -21,7 +21,31 @@ type decValueRenderer struct{}
 var _ ValueRenderer = decValueRenderer{}
 
 func (vr decValueRenderer) Format(_ context.Context, v protoreflect.Value) ([]Screen, error) {
-	formatted, err := math.FormatDec(v.String())
+	decStr := v.String()
+
+	// If the decimal doesn't contain a point, we assume it's a badly formatted
+	// value. So we assume it's a decimal with 18 decimal places and add the decimal
+	// in the right place.
+	if !strings.Contains(decStr, ".") {
+		isNeg := false
+		if strings.HasPrefix(decStr, "-") {
+			decStr = strings.TrimPrefix(decStr, "-")
+			isNeg = true
+		}
+
+		if len(decStr) < 19 {
+			decStr = "0." + strings.Repeat("0", 18-len(decStr)) + decStr
+		} else {
+			whole, decimal := decStr[:len(decStr)-18], decStr[18:]
+			decStr = whole + "." + decimal
+		}
+
+		if isNeg {
+			decStr = "-" + decStr
+		}
+	}
```

Comment:
Implementation LGTM. I'd pull this conversion out and heavily unit-test it separately. But that's personal preference.

When I suggested a string based solution I was not aware how it would look in Go. But there does not even seem to be string padding in the standard library. I don't see a way to improve the code right now.

Reply:
I'll comment here how it would look like if done using the math package. It's a bit more compact

Reply:
I think code readability is more important here, I would go with math if it's more compact

Reply:
Switched to math

Reply:
```
...Options (1/1)xWeightedVoteOption object�fOptionoVOTE_OPTION_YES�fWeighta1�nEnd of Options�nEnd of Message�iGas limitg200'000...
```
![WhatsApp Image 2023-02-23 at 3 17 02 PM](https://user-images.githubusercontent.com/14063057/220995571-960da3ec-4da1-4a51-a248-f5177d1456fe.jpeg)

Looks fine 👌 Also for bank send it looks good.